startswithInsensitive.go

```diff
@@ -0,0 +1,40 @@
+package tests
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+// $startsWith
+type startsWithInsensitiveTest struct {
+	expectedStart string
+}
+
+func NewStartsWithInsensitiveTest(input interface{}) (Test, error) {
+	castedString, ok := castString(input)
+	if !ok {
+		return nil, fmt.Errorf("value for $startsWith is not a string (%#v %v)", input, reflect.TypeOf(input))
+	}
+	return &startsWithInsensitiveTest{
+		expectedStart: castedString,
+	}, nil
+}
+
+func (swt *startsWithInsensitiveTest) Run(input interface{}) error {
+	castedString, ok := castString(input)
+	if !ok {
+		return fmt.Errorf("input (%#v %v) isn't a string", input, reflect.TypeOf(input))
+	}
+
+	if !strings.HasPrefix(strings.ToLower(castedString), strings.ToLower(swt.expectedStart)) {
```

Comment:
`strings.ToLower` doesn't do unicode case-folding which is used in the other test added in this PR
I think we should be consistent here

Reply:
fwiw I believe we do not support title case characters (see https://www.compart.com/en/unicode/category/Lt) so the simple ToLower/ToUpper should be fine.

Reply:
if we use ToLower here, then `EqualFold` shouldn't be used in the other test, as the implementation might lead to different results in edge cases